src/packages/signer/src/components/RicardianContract.svelte

```diff
@@ -0,0 +1,60 @@
+<script lang="ts">
+    import ricardianTemplate from '../../../../../assets/ricardian-contracts/index.html?raw'
+
+    export let action
+    export let abi
+    export let transaction
+
+    if (!action || !abi || !transaction) {
+        throw new Error('RicardianContract - Missing action, ABI or transaction')
+    }
+
+    const relevantTransaction = {
+        ...transaction,
+        actions: [{
+            ...action,
+            data: action.decodeData(abi)
+        }],
+    }
+
+    const ricardianScript =  `
+      var contractContainer = document.getElementById("contract-container");
+      var factory = new ContractTemplateToolkit.RicardianContractFactory();
+
+      var config = {
+        abi: ${JSON.stringify(abi)},
+        transaction: ${JSON.stringify(relevantTransaction)},
+        actionIndex: 0,
+        maxPasses: 3,
+        allowUnusedVariables: false,
+      };
+
+      var ricardianContract = factory.create(config);
+      var html = ricardianContract.getHtml();
+
+      contractContainer.innerHTML = html;
+    `
+
+    const ricardianTemplateParts = ricardianTemplate.split("/body>")
+
+    const ricardianTemplateWithScript = `${ricardianTemplateParts[0]}/body>\n<script>${ricardianScript}\n<` + `/script>\n${ricardianTemplateParts[1]}`
```

Comment:
What's with the string concatenation in the middle of this string?

Reply:
I did that because my IDE would freak out when a single `<script />` tag would be present in the middle of the string.